assets/js/index.js

```diff
@@ -0,0 +1,151 @@
+  const minMax = document.getElementById('minMax');
+  const current = currentTemp.textContent.split(' ')[0];
+  const min = minMax.textContent.split(' ')[1];
+  const max = minMax.textContent.split(' ')[5];
+  [...document.getElementsByClassName('unit')].forEach((div) => {
+    div.classList.toggle('active');
+  });
+  currentTemp.textContent = `${convert(current, unit)} °${unit}`;
+  minMax.textContent = `Min. ${convert(min, unit)} °${unit} | Max. ${convert(max, unit)} °${unit}`;
+};
+
+const addWeatherResults = (response, parent) => {
+  const div = document.createElement('div');
+  div.classList.add('card');
+  const title = document.createElement('h2');
+  const icon = document.createElement('img');
+  const temp = document.createElement('big');
+  temp.setAttribute('id', 'currentTemp');
+  const minMax = document.createElement('p');
+  minMax.setAttribute('id', 'minMax');
+  const description = document.createElement('h3');
+  const more = document.createElement('h6');
+  const wind = document.createElement('p');
+  const pressure = document.createElement('p');
+  const humidity = document.createElement('p');
+
+  icon.src = `http://openweathermap.org/img/wn/${response.weather[0].icon}@2x.png`;
+  icon.alt = `${response.weather[0].description}`;
+
+  let str = '';
+  for (let i = 1; i < response.weather.length; i += 1) {
+    i === 1 ? (str += 'with ') : (str += ', ');
+    str += `${response.weather[i].description}`;
+  }
+
+  [
+    [title, `${response.name}, ${response.sys.country}`],
+    [temp, `${parseInt(response.main.temp)} °C`],
+    [
+      minMax,
+      `Min. ${parseInt(response.main.temp_min)} °C | Max. ${parseInt(response.main.temp_max)} °C`,
+    ],
+    [description, `${response.weather[0].description}`],
+    [more, `${str}`],
+    [humidity, `Humidity: ${response.main.humidity} %`],
+    [pressure, `Pressure: ${response.main.pressure} hpa`],
+    [wind, `Wind: ${response.wind.speed} m/s`],
+  ].forEach((item) => {
+    item[0].textContent = item[1];
+  });
+
+  [title, icon, temp, minMax, description, more, wind, pressure, humidity].forEach((item) => {
+    if (item.tagName === 'IMG' || item.textContent !== '') div.appendChild(item);
+  });
+
+  parent.appendChild(div);
+};
+
+const addUnitToggler = (parent) => {
+  const div = document.createElement('div');
+  div.classList.add('toggler');
+  const c = document.createElement('div');
+  c.textContent = '°C';
+  c.value = 'C';
+  c.classList.add('unit');
+  c.classList.add('active');
+  c.addEventListener('click', e => toggleUnit(e, 'C'), false);
+  const f = document.createElement('div');
+  f.value = 'F';
+  f.textContent = '°F';
+  f.classList.add('unit');
+  f.addEventListener('click', e => toggleUnit(e, 'F'), false);
+  // unit === 'C' ? c.classList.add('active') : f.classList.add('active');
+  div.appendChild(c);
+  div.appendChild(f);
+  parent.appendChild(div);
+};
+
+const fetchCities = (searchTerm) => {
```

Comment:
To make our data-fetching logic independent what we wanna do is separate the DOM related stuff from it. Since `fetchCities` returns a `Promise` we can then utilize that returned promise in another function that processes the fetched data (or handles any error).

```javascript
cnst fetchCities = (searchTerm) => {
  const url = 'https://api.openweathermap.org/data/2.5/weather?q=${searchTerm}&units=metric&appid=b8d08b4b5bc310505709e7342891ec46';

  return fetch(url, {mode: 'cors'}).then(res => res.json()).catch(err => console.log("Do other error handling stuff here", err));
}
```

Then we can utilize this function as such:

```javascript
const processFetchedData = searchTerm => {
  fetchCities(searchTerm).then(data => // do something now with the data).catch(err => //do some error handling)
}
```

It might seem like adding a step that you could have just chained to the original `fetchCities` function. But generally it's a good practice to keep your data fetching from the API separate from processing it. The main reason, in my experience, is say, the API changes or you need to change how you handle things. This keeps concerns separate. 

Also, what do you think about using `async` and `await` to make things cleaner? :) And maybe you can consider using [`axios`](https://github.com/axios/axios) - it's an awesome package to use in lieu of `fetch` for data fetching.